kv/memberlist/dnsprovider.go

```diff
@@ -0,0 +1,15 @@
+package memberlist
+
+import (
+	"context"
+)
+
+type DNSProvider interface {
```

Comment:
nit: I realize this exists so that we don't need to depend on Thanos, but could we include some examples of input addresses and results to help people that want to create an implementation of this?

Reply:
See my comment below.